src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -493,7 +495,8 @@ protected Path search()
             handleDebugOptions(currentNode);
             currentNode.setClosed();
 
-            if (isAtDestination(currentNode))
+            if ((!xzRestricted || (currentNode.pos.getX() >= minX && currentNode.pos.getX() <= maxX && currentNode.pos.getZ() >= minZ
+                && currentNode.pos.getZ() <= maxZ)) && isAtDestination(currentNode))
```

Comment:
moved the xz restriction here, this allows to have start outside the area

Reply:
Yeah but this also makes it it so that everything outside of the requested area is searched for a path. Which is not actually what you always want.

Document this new behaviour better.

Reply:
True, there are two things to consider:
a) for this open-restricted variant, when to end search if at least eg. 90% entire area was searched
b) actually having option for closed-restricted variant - which will simply end particularly not searched for things like two levelled terrain with unjumpable barrier, or just fence line